mlir/include/mlir/Dialect/ArmNeon/ArmNeon.td

```diff
@@ -120,6 +120,99 @@ def SdotOp : ArmNeon_OverloadedOperandsWithOneResultIntrOp<"sdot",[1], [
     "$a `,` $b `,` $c attr-dict `:` type($b) `,` type($c) `to` type($res)";
   }
 
+def SmmlaOp : ArmNeon_OverloadedOperandsWithOneResultIntrOp<"smmla",[1], [
+                Pure,
+                AllTypesMatch<["b", "c"]>,
+                AllTypesMatch<["a", "res"]>,
```

Comment:
Please use descriptive variable names like `src1`/`src` and `acc/res`. For example:
* https://github.com/llvm/llvm-project/blob/e3172e841899779caf461cee97826940dca806e7/mlir/include/mlir/Dialect/ArmSVE/IR/ArmSVE.td#L153-L157